From 40ed277b33e92200fd20248890444d7fac723f6b Mon Sep 17 00:00:00 2001
From: Arun Kumar <arunkumar.ev@adlinktech.com>
Date: Mon, 11 May 2020 12:28:09 +0530
Subject: [PATCH] IMX6R2: Do not disable RTC output clock

Do not disable CLKOUT line from RTC PCF8563. Set
CLKOUT register 0x0D to 0x80. Also default CPU RTC
driver is disabled in config

Signed-off-by: Arun Kumar <arunkumar.ev@adlinktech.com>

diff --git a/drivers/rtc/rtc-pcf8563.c b/drivers/rtc/rtc-pcf8563.c
index 8c836c51a508..28cefdc7b139 100644
--- a/drivers/rtc/rtc-pcf8563.c
+++ b/drivers/rtc/rtc-pcf8563.c
@@ -202,7 +202,7 @@ static irqreturn_t pcf8563_irq(int irq, void *dev_id)
 static int pcf8563_get_datetime(struct i2c_client *client, struct rtc_time *tm)
 {
 	struct pcf8563 *pcf8563 = i2c_get_clientdata(client);
-	unsigned char buf[9];
+	unsigned char buf[9], buf1;
 	int err;
 
 	err = pcf8563_read_block_data(client, PCF8563_REG_ST1, 9, buf);
@@ -244,6 +244,11 @@ static int pcf8563_get_datetime(struct i2c_client *client, struct rtc_time *tm)
 		tm->tm_sec, tm->tm_min, tm->tm_hour,
 		tm->tm_mday, tm->tm_mon, tm->tm_year, tm->tm_wday);
 
+	err = pcf8563_read_block_data(client, PCF8563_REG_CLKO, 1, &buf1);
+	pr_debug("Reg 0x0D value - %0x\n", buf1);
+	if (err)
+		return err;
+
 	return 0;
 }
 
@@ -488,7 +493,8 @@ static int pcf8563_clkout_prepare(struct clk_hw *hw)
 
 static void pcf8563_clkout_unprepare(struct clk_hw *hw)
 {
-	pcf8563_clkout_control(hw, 0);
+	//TBD: Do not disable CLKOUT line fow now
+	//pcf8563_clkout_control(hw, 0);
 }
 
 static int pcf8563_clkout_is_prepared(struct clk_hw *hw)
@@ -544,6 +550,12 @@ static struct clk *pcf8563_clkout_register_clk(struct pcf8563 *pcf8563)
 	if (!IS_ERR(clk))
 		of_clk_add_provider(node, of_clk_src_simple_get, clk);
 
+	/*Enable CLKOUT line*/
+	buf = 0x80;
+	ret = pcf8563_write_block_data(client, PCF8563_REG_CLKO, 1, &buf);
+	if (ret < 0)
+		return ERR_PTR(ret);
+
 	return clk;
 }
 #endif
-- 
2.17.1

